### PR TITLE
Add Transformer to filter sensitive data 

### LINKED ISSUE #1: Filter sensitive data feature
This is self-explanatory. If there are sensitive API keys being sent, you don't want to commit those in your respository (especially in a public repo). Perhaps `VcrClient` could also take in a `Transformer` class of some kind which does post processing before saving to disk.

I imagine you should be able to transform both requests and responses. Not only requests.

### PR DESCRIPTION
**Pull request by exoego**
Closes #1

This PR adds `RequestTransformer` and `ResposneTransformer`.
As name suggested, they transform `VcrRequest` and `VcrResponse`.

This partially adresses #10: in-memory storage `newlyRecorded` is encapsulated, to ensure that data are transformed before writing to `newlyRecorded`.

`VcrRecord` have `request`, `response` and `recordedAt` fields right now.
If one want to transform `recordedAt` and future fields, each fields requires their own transformers.
I think it is a bit verbose, so maybe `RecordTransformer` is better ?

(Potential usecase: normalize the timezone of `recordedAt`, etc)

### CODE REVIEW

File: core/src/main/scala/bekindrewind/RecordOptions.scala
Comment (reibitto) [inline core/src/main/scala/bekindrewind/RecordOptions.scala:26]: It's a stylistic choice, but I'm okay with just using `identity` and getting rid of these `noop` aliases.

File: core/src/main/scala/bekindrewind/VcrClient.scala
Comment (reibitto) [inline core/src/main/scala/bekindrewind/VcrClient.scala:42]: I understand the meaning behind it, but this feels like an awkward name to me because "current" and "new" are contradictory. Maybe `getNewlyRecorded` instead? Or just `newlyRecorded` and then rename the `AtomicReference` one `newlyRecordedRef` or something like that? Just an idea.

**Review by reibitto — CHANGES_REQUESTED** [review]
Thanks for yet another PR 🙂

Yeah, that's a good point about the transformer. I think I like `RecordTransformer` for the reasons you mentioned. It being more future-proof is nice too.

**Comment by reibitto** [comment]
Also, I just realized I'm using the word `Record` in 2 different ways and that might be confusing 😅 Maybe I should rename `VcrRecord` to `VcrEntry` or something. And then only use the word `Record` to refer to the record action ⏺

**Comment by reibitto** [comment]
I added the ability to combine multiple matchers together in #16. I'll hold off merging that until this PR gets merged though. I don't want to cause a bunch of merge conflicts for you.

I think this means that I'll be moving your RecordTransformer to `VcrMatcher` as well. Because it makes sense that you'll want different transformers for different websites/APIs. In fact, maybe the entire `RecordOptions` should go inside `VcrMatcher`.

**Comment by exoego** [comment]
@reibitto PR Updated 🙇

**Review by reibitto — APPROVED** [review]
Thanks!

**Pull request merged**
